test/procedure.test.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+var Database = require('../lib/database');
+var drivers = require('./drivers');
+var mssql = require('mssql');
+var should = require('should');
+
+var getDriver = function(name, opts) {
+  return new Database(name, opts);
+};
+
+var driver = drivers[1];
+var db = getDriver(driver.name, driver.opts);
+
+before(function(done) {
+  db.execute(driver.dropSP1)
+    .then(function(){
+      db.execute(driver.createSP1);
+    });
+  db.execute(driver.dropSP2)
+    .then(function(){
+      db.execute(driver.createSP2);
+    });
+  db.execute(driver.dropSP3)
+    .then(function(){
+      db.execute(driver.createSP3);
+    });
+  done();
```

Comment:
You're calling done here before the completion of the db.exeutes.
The done() should be a .then() block